### PR TITLE
[WNMGDS-1878/1879/1880] Table component React Testing Library updates

### PR DESCRIPTION
**Pull request by forrestbaer**
## Summary

- Updated tests to use React Testing Library for `Table` `TableBody` `TableCaption` `TableRow` `TableHeader` `TableCell`
- Added eslint plugin for testing library (can pull if too much, seems like a nice to have but could do a separate PR for it)

## How to test

1. `yarn install && yarn test`

### If this is a change to code:

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[ticket] title`
- [x] Ran `yarn type-check && yarn lint && yarn test` with no errors
- [x] Updated unit test snapshots and loki reference images **if necessary** with `yarn loki` && `yarn update-snapshots`

### CODE REVIEW

File: packages/design-system/src/components/Table/Table.test.tsx [hunk 1]
```diff
@@ -1,149 +1,70 @@
...
+
 import Table from './Table';
 import TableCaption from './TableCaption';
+// import TableBody from './TableBody';
 
-const defaultCaptionChildren = 'Foo';
-const tableCaption = <TableCaption>{defaultCaptionChildren}</TableCaption>;
-
-function render(customProps = {}, children = tableCaption, deep = false) {
+const makeTable = (customProps = {}) => {
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/Table.test.tsx:8]: ron-approves

File: packages/design-system/src/components/Table/Table.test.tsx [hunk 2]
```diff
@@ -1,149 +1,70 @@
...
   const props = Object.assign({}, customProps);
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/Table.test.tsx:9]: This line doesn't seem necessary. Could just spread the props directly on line 12

File: packages/design-system/src/components/Table/Table.test.tsx [hunk 3]
```diff
@@ -1,149 +1,70 @@
...
-    });
-
-    it('applies responsive stacked table ContextProvider', () => {
-      const { wrapper } = render({ stackable: true, warningDisabled: true });
-      const context = wrapper.find('ContextProvider');
-
-      expect(context.prop('value')).toMatchObject({ stackable: true, warningDisabled: true });
+  it('applies scrollable prop to table', () => {
+    makeTable({ scrollable: true });
+    const scrollableRegion = screen.getByRole('region');
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/Table.test.tsx:55]: Whoa, this version seems a lot better 👍

File: packages/design-system/src/components/Table/Table.test.tsx [hunk 4]
```diff
@@ -1,149 +1,70 @@
-import { mount, shallow } from 'enzyme';
 import React from 'react';
+import { render, screen } from '@testing-library/react';
+
 import Table from './Table';
 import TableCaption from './TableCaption';
+// import TableBody from './TableBody';
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/Table.test.tsx:6]: 💥 code comment

File: packages/design-system/src/components/Table/TableRow.test.tsx [hunk 5]
```diff
@@ -13,33 +14,20 @@ function render(customProps = {}) {
...
-    expect(tableRow).toHaveLength(1);
-  });
-
-  it('sets role="row"', () => {
-    const { wrapper } = render();
-    const tableRow = wrapper.find('tr');
-
-    expect(tableRow.prop('role')).toBe('row');
+    makeTableRow();
+    expect(screen.getByRole('row')).toBeInTheDocument();
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/TableRow.test.tsx:23]: Nice and succinct 👌

File: packages/design-system/src/components/Table/__snapshots__/Table.test.tsx.snap [hunk 6]
```diff
@@ -1,141 +1,36 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Table table caption scrollable true applies scroll table wrapper and classes 1`] = `
```
Comment (pwolfert) [inline packages/design-system/src/components/Table/__snapshots__/Table.test.tsx.snap:3]: I love shrinking snapshot tests

**Review by pwolfert — APPROVED** [review]
Nice tests!

File: packages/design-system/src/components/Table/Table.test.tsx (hunk 1 above)
Comment (pwolfert) [reply packages/design-system/src/components/Table/Table.test.tsx:8]: That is, he approves of _make table_

File: packages/design-system/src/components/Table/Table.test.tsx (hunk 2 above)
Comment (forrestbaer) [reply packages/design-system/src/components/Table/Table.test.tsx:9]: Good call.

File: packages/design-system/src/components/Table/Table.test.tsx (hunk 1 above)
Comment (forrestbaer) [reply packages/design-system/src/components/Table/Table.test.tsx:8]: Hahaha nice

**Pull request merged**